stellarphot/core.py

```diff
@@ -116,49 +120,105 @@ class BaseEnhancedTable(QTable):
+        names to the desired names BEFORE the validation is performed.
     """
 
-    def __init__(self, table_description, data):
+    def __init__(self, table_description, data, colname_map=None):
+        orig_data = data.copy()
+
         # Confirm a proper table description is passed (that is dict-like with keys and
         # values)
         try:
             self._table_description = {k: v for k, v in table_description.items()}
         except AttributeError:
-            raise TypeError("You must provide a dict as table_description (it is type "
-                + f"{type(self._table_description)}).")
+            raise TypeError("You must provide a dict as table_description (input "
+                f"table_description is type {type(self._table_description)}).")
 
         # Build the data table
         if not isinstance(data, Table):
-            raise TypeError("You must provide an astropy QTable as data (it is type "
-                            + f"{type(data)}).")
+            raise TypeError("You must provide an astropy Table as data (input data is "
+                            f"type {type(data)}).")
         else:
-            # Check the format of the data table matches the table_description by
-            # checking each column listed in table_description exists and is the
-            # correct units.
-            # NOTE: This ignores any columns not in the table_description, it
-            # does not remove them.
-            for this_col, this_unit in self._table_description.items():
-                if this_unit is not None:
-                    # Check type
-                    try:
-                        if data[this_col].unit != this_unit:
-                            raise ValueError(f"data['{this_col}'] is of wrong unit "
-                                             + f"(should be {this_unit} but reported "
-                                             + f"as {data[this_col].unit}).")
-                    except KeyError:
-                        raise ValueError(f"data['{this_col}'] is missing from input "
-                                         + "data.")
-
-            # Using the BaseTimeSeries class to handle the data table means required
+            # Rename columns before validation (if needed)
+            if colname_map is not None:
+                # Confirm a proper colname_map is passed
+                try:
+                    self._colname_map = {k: v for k, v in colname_map.items()}
+                except AttributeError:
+                    raise TypeError("You must provide a dict as table_description "
+                                    "(input table_description is type "
+                                    f"{type(self._table_description)}).")
+
+                self._update_colnames(self._colname_map,orig_data)
+
+            # Validate the columns
+            self._validate_columns(orig_data, self._table_description)
```

Comment:
Suggest changing `self._validate_columns` so that it access `self._table_description` directly.

```suggestion
            self._validate_columns(orig_data)
```

Reply:
Fixed. At one point I though each of the derived classes would have to have access to validation.